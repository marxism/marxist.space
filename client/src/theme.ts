const constants = {
  error: '#f5222d',
}

const dark = {
  ...constants,
  normalText: '#ffffff',
  mutedText: '#b0b8bf',
  border: '#333333',
  accent: '#33a0ff',
  pageBackground: '#1b1b1b',
  foreground: '#262626',
  activeBackground: '#333333',
  inputBackground: '#212121',
  shadow: 'rgba(0, 0, 0, 0.4)',
}

const light = {
  ...constants,
  normalText: '#353c46',
  mutedText: '#64717d',
  border: '#ebedf0',
  accent: '#1890ff',
  pageBackground: '#f4f6f8',
  foreground: '#ffffff',
  activeBackground: '#fafafa',
  inputBackground: '#fcfcfc',
  shadow: 'rgba(0, 0, 0, 0.05)',
}

const theme = (isDark) => (isDark ? dark : light)

export default theme
