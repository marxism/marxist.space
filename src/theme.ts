const theme = {
  error: '#f5222d',
  normalText: '#353c46',
  mutedText: '#64717d',
  border: '#ebedf0',
  accent: '#1890ff',
  pageBackground: '#f4f6f8',
  foreground: '#ffffff',
  activeBackground: '#fafafa',
  inputBackground: '#fcfcfc',
  shadow: 'rgba(0, 0, 0, 0.05)',
}

export default theme
