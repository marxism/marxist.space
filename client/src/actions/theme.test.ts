/* eslint-env jest */

import * as themeActions from './theme'

describe('theme actions', () => {
  test('toggleDarkTheme', () => {
    expect(themeActions.toggleDarkTheme()).toMatchSnapshot()
  })
})
