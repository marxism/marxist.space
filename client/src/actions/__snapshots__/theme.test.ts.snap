// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme actions toggleDarkTheme 1`] = `
Object {
  "type": "TOGGLE_DARK_THEME",
}
`
